baseplate/server/__init__.py

```diff
@@ -77,6 +77,7 @@ def read_config(config_file, server_name, app_name):
     # that config.Percent works more naturally (no escaping %).
     parser = configparser.RawConfigParser()
     parser.readfp(config_file)  # pylint: disable=deprecated-method
+    config_file.close()
```

Comment:
It's opened in `parse_args` which clearly can't close it because then it'd never go anywhere, and then `parse_args` + `read_config` are called in multiple entrypoints so we'd have to repeat the `.close()` everywhere. Definitely icky.

Reply:
ahh, a 🔕 but maybe there's a way to give a single function that calls both of those and handles opening and cleaning up the file?

Reply:
from what i can tell, the only time that this would be a problem is if someone is manually wiring things up vs. calling the `load_and_run_*` funcitons -- if that's the case, maybe having the `load_and_run_*` functions close the file handle, that way `read_config` doesn't have this side effect? The `load_and_run_` functions are where all the side effects are anyways so it makes a certain amount of sense to do the filehandle lifecycle there.

Reply:
Yeah, I hear you guys. I'll do a cleanup pass on this file separately. This commit is tangential to the rest of this.

Reply:
sgtm

Reply:
:bike: :house: